JoinColony/colonyDapp
# Add Edit Decision Modal

## Description

This PR wires up the "Edit" button on the Decisions Preview page with the `NewDecisionDialog` modal.

To test: 
* Go to /decisions/preview and click "Edit". Form content should be prefilled and submit button text should say "Update".
* Then to be sure go back to /decisions and click "New Decision". Form should be empty and button should say "Preview". 

**Changes** 🏗

* "Edit" onClick now opens `NewDecisionDialog`
* NewDecisionDialog takes optional "Title" and "Content" props so it can display the data entered previously when opened from Preview page

## Screenshot 

When being edited, if the form values don't change, the form will be disabled (because it wont be dirty), which I think is the correct behaviour.

![editModal](https://user-images.githubusercontent.com/64402732/185403843-55b5b7cb-79f8-47da-8e09-0a9d037b8aa4.png)

Resolves #3710
